src/layer_shell_v1.cpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright © 2018-2019 Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: William Wold <william.wold@canonical.com>
+ */
+
+#include "layer_shell_v1.h"
+
+wlcs::LayerSurfaceV1::LayerSurfaceV1(
+    wlcs::Client& client,
+    wlcs::Surface& surface,
+    zwlr_layer_shell_v1_layer layer,
+    wl_output* output,
+    const char *_namespace)
+    : client{client}
+{
+    layer_surface = zwlr_layer_shell_v1_get_layer_surface(
+        client.layer_shell_v1(),
+        surface,
+        output,
+        layer,
+        _namespace);
+    static struct zwlr_layer_surface_v1_listener const listener {
+        +[](void* data,
```

Comment:
What's the "+" prefix doing here? Is this to force decay to a function pointer?

Reply:
Correct

Reply:
The decay happens anyway, I don't think the "+" makes anything clearer to the reader.

Reply:
Strange, I thought those pluses were required. Maybe they were in C++11 or something. Anyway, removed.